tensorflow/python/ops/array_ops.py

```diff
@@ -986,6 +987,62 @@ def pad(tensor, paddings, mode="CONSTANT", name=None):  # pylint: disable=invali
   raise ValueError("Unknown padding mode: %s" % mode)
 
 
+def meshgrid(*args, indexing="xy", name="meshgrid"):
+  """Broadcasts parameters for evaluation on an N-D grid.
+
+  Given N one-dimensional coordinate arrays `*args`, returns a list `outputs`
+  of N-D coordinate arrays for evaluating expressions on an N-D grid.
+
+  For example:
+
+  Calling `X, Y = meshgrid(x, y)` with the tensors
+  ```prettyprint
+    x = [1, 2, 3]
+    y = [4, 5, 6]
+  ```
+  results in
+  ```prettyprint
+    X = [[1, 1, 1],
+         [2, 2, 2],
+         [3, 3, 3]]
+    Y = [[4, 5, 6],
+         [4, 5, 6],
+         [4, 5, 6]]
+  ```
+
+  Args:
+    *args: `Tensor`s with rank 1
+
+  Returns:
+    outputs: A list of N `Tensor`s with rank N
+
+  """
+  if indexing not in ("xy", "ij"):
+    raise ValueError("indexing parameter must be either 'xy' or 'ij'")
+
+  num_inputs = len(args)
+
+  if num_inputs > 5:
+    raise ValueError("meshgrid only supports up to 5 dimensions")
```

Comment:
I think it's fine to drop this assertion and let tile break or not as it pleases.  In particular, if tile becomes less constrained later it'd be nice if this routine didn't need to change.